tasks/setup.py

```diff
@@ -277,6 +266,353 @@ def setupPulumiConfig(config):
         warn(f"Expecting one of [true, false], got {logs_to_std_err}")
 
 
+def resolve_keypair_opts(
+    config: Config,
+    keypair_name: Optional[str] = None,
+    key_type: Optional[str] = None,
+    key_format: Optional[str] = None,
+    private_key_path: Optional[Path | str] = None,
+    public_key_path: Optional[Path | str] = None,
+    require_key_type: Optional[bool] = False,
+    require_keyfile_exists: Optional[bool] = False,
+):
+    """
+    Resolve key pair options, in decreasing order of priority:
+    - user input (parameters)
+    - local config
+    - defaults
+
+    Returns a dict with the resolved values.
+    """
+    if config.configParams is None or config.configParams.aws is None:
+        raise Exit("Config is missing aws section")
+    awsConf = config.configParams.aws
+
+    # defaults
+    if awsConf.keyPairName:
+        default_keypair_name = awsConf.keyPairName
+    else:
+        default_keypair_name = getpass.getuser()
+    if awsConf.privateKeyPath:
+        default_private_key_path = awsConf.privateKeyPath
+    else:
+        default_private_key_path = None
+    if awsConf.publicKeyPath:
+        default_public_key_path = awsConf.publicKeyPath
+    else:
+        default_public_key_path = None
+
+    # ask for missing values
+    if not keypair_name:
+        keypair_name = ask(f"🔑 Key pair name (default: {default_keypair_name}): ")
+        if not keypair_name:
+            keypair_name = default_keypair_name
+    if not key_type and require_key_type:
+        warn('Creating Windows VMs requires "rsa" key type')
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
I would create only `rsa` keys as Windows VMs support only `rsa` keys and other VMs support it
I would avoid warning if possible, except on issues